Library/Homebrew/dev-cmd/test.rb

```diff
@@ -93,9 +93,14 @@ def test
             exec(*args)
           end
         end
-      rescue ::Test::Unit::AssertionFailedError => e
+      rescue ChildProcessError => e
         ofail "#{f.full_name}: failed"
-        puts e.message
+        case e.inner["json_class"]
```

Comment:
I could add `ChildProcessError.inner_class` that just calls `const_get` on `inner["json_class"]`, resulting in something like this:

```ruby

if e.inner_class == ::Test::Unit::AssertionFailedError
  puts e.inner["m"]
else
  puts e.inner_class, e.backtrace
end
```

How does that sound?

Reply:
Yeh, I think that'd be nicer 👍. I do wonder if this couldn't be done at a higher level somewhere (`Utils.safe_fork`?) so that it doesn't need to be handled as JSON in so many places.